runtime/cudaq/platform/default/rest_server/helpers/RestRemoteServer.cpp

```diff
@@ -0,0 +1,268 @@
+
+  virtual void handleRequest(std::size_t reqId,
+                             cudaq::ExecutionContext &io_context,
+                             const std::string &backendSimName,
+                             std::string_view ir, std::string_view kernelName,
+                             void *kernelArgs, std::uint64_t argsSize,
+                             std::size_t seed) override {
+    void *handle = loadNvqirSimLib(backendSimName);
+    if (seed != 0)
+      cudaq::set_random_seed(seed);
+    auto &platform = cudaq::get_platform();
+    platform.set_exec_ctx(&io_context);
+
+    auto &requestInfo = m_codeTransform[reqId];
+
+    if (requestInfo.format == cudaq::CodeFormat::LLVM)
+      cudaq::invokeWrappedKernel(ir, std::string(kernelName), kernelArgs,
+                                 argsSize);
+    else
+      invokeMlirKernel(m_mlirContext, ir, requestInfo.passes,
+                       std::string(kernelName));
+    platform.reset_exec_ctx();
+    dlclose(handle);
+  }
+
+private:
+  std::unique_ptr<ExecutionEngine>
+  jitMlirCode(ModuleOp currentModule, const std::vector<std::string> &passes,
+              const std::vector<std::string> &extraLibPaths = {}) {
+    cudaq::info("Running jitCode.");
+    auto module = currentModule.clone();
+    ExecutionEngineOptions opts;
+    opts.transformer = [](llvm::Module *m) { return llvm::ErrorSuccess(); };
+    opts.enableObjectDump = true;
+    opts.jitCodeGenOptLevel = llvm::CodeGenOpt::None;
+    SmallVector<StringRef, 4> sharedLibs;
+    for (auto &lib : extraLibPaths) {
+      cudaq::info("Extra library loaded: {}", lib);
+      sharedLibs.push_back(lib);
+    }
+    opts.sharedLibPaths = sharedLibs;
+
+    auto ctx = module.getContext();
+    PassManager pm(ctx);
+    std::string errMsg;
+    llvm::raw_string_ostream os(errMsg);
+    const std::string pipeline =
+        std::accumulate(passes.begin(), passes.end(), std::string(),
+                        [](const auto &ss, const auto &s) {
+                          return ss.empty() ? s : ss + "," + s;
+                        });
+    if (failed(parsePassPipeline(pipeline, pm, os)))
+      throw std::runtime_error(
+          "Remote rest platform failed to add passes to pipeline (" + errMsg +
+          ").");
+
+    if (failed(pm.run(module)))
+      throw std::runtime_error(
+          "Remote rest platform: applying IR passes failed.");
+
+    cudaq::info("- Pass manager was applied.");
+
+    opts.llvmModuleBuilder =
+        [](Operation *module,
+           llvm::LLVMContext &llvmContext) -> std::unique_ptr<llvm::Module> {
+      llvmContext.setOpaquePointers(false);
+      auto llvmModule = translateModuleToLLVMIR(module, llvmContext);
+      if (!llvmModule) {
+        llvm::errs() << "Failed to emit LLVM IR\n";
+        return nullptr;
+      }
+      ExecutionEngine::setupTargetTriple(llvmModule.get());
+      return llvmModule;
+    };
+
+    cudaq::info(" - Creating the MLIR ExecutionEngine");
```

Comment:
nit: why the space before the `-`? The info message below does not have that space.

Reply:
Fixed in https://github.com/NVIDIA/cuda-quantum/pull/1012/commits/509ba068d304bdde90afa7dc086e14dd063ed95d